storage/rocksdb/ha_rocksdb.cc

```diff
@@ -3090,9 +3164,73 @@ static int rocksdb_close_connection(handlerton *const hton, THD *const thd) {
     delete tx;
     tx = nullptr;
   }
+  const char *checkpoint_dir = get_ha_data(thd)->get_checkpoint_dir();
+  if (checkpoint_dir != nullptr) {
+    rocksdb_remove_checkpoint(checkpoint_dir);
+    get_ha_data(thd)->set_checkpoint_dir(nullptr);
+  }
+  if (get_ha_data(thd)->disable_file_deletions) {
+    rdb->EnableFileDeletions(false);
+  }
+  destroy_ha_data(thd);
+  return HA_EXIT_SUCCESS;
+}
+
+static int rocksdb_create_temporary_checkpoint_validate(
+    my_core::THD *const thd, my_core::SYS_VAR *const /* unused */,
+    void *const save, my_core::st_mysql_value *const value) {
+  DBUG_ASSERT(rdb != nullptr);
+  DBUG_ASSERT(thd != nullptr);
+
+  char buf[FN_REFLEN];
+  int len = sizeof(buf);
+
+  const char *current_checkpoint_dir = get_ha_data(thd)->get_checkpoint_dir();
+  const char *new_checkpoint_dir = value->val_str(value, buf, &len);
+
+  if (current_checkpoint_dir != nullptr && new_checkpoint_dir != nullptr) {
+    std::string errmsg = "Checkpoint already exists in directory '" +
```

Comment:
Local var which is never used?..

Reply:
I lost the following line somewhere between iterations:
```
    my_error(ER_GET_ERRMSG, MYF(0), HA_ERR_ROCKSDB_STATUS_INVALID_ARGUMENT,
             errmsg.c_str(), rocksdb_hton_name);
```